src/components/Deployment/Deployment.tsx

```diff
@@ -0,0 +1,229 @@
+import React, { useState, useEffect } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { AccountStates } from 'etherspot';
+
+// icons
+import { IoMdCheckmark } from 'react-icons/io';
+import { CgSandClock } from 'react-icons/cg';
+
+// components
+import Card from '../Card/Card';
+import { RoundedImage } from '../Image';
+import { useEtherspot } from '../../hooks';
+import Modal from '../Modal/Modal';
+import { Paragraph } from '../Text';
+import { PrimaryButton, SecondaryButton } from '../Button';
+import ErrorMessage from '../Error/ErrorMessage';
+
+// utils
+import { supportedChains } from '../../utils/chain';
+import { deployAccount } from '../../utils/transaction';
+import { Theme } from '../../utils/theme';
+
+const Wrapper = styled.div`
+  margin-top: 10px;
+`;
+
+const Section = styled.div`
+  display: flex;
+  background: ${({ theme }) => theme.color.background.topMenu};
+  margin-top: 5px;
+  padding: 15px;
+  border-radius: 10px;
+  font-weight: bold;
+`;
+
+const Header = styled.div`
+  display: block;
+`;
+
+const Body = styled.div`
+  display: block;
+`;
+
+const Label = styled.label`
+  display: inline-block;
+  margin-bottom: 14px;
+  font-size: 14px;
+`;
+
+const AuthLabel = styled.div`
+  background-color: ${({ theme }) => theme.color.background.button};
+  border-radius: 10px;
+  font-weight: bold;
+  padding: 5px;
+  position: absolute;
+  right: 28px;
+`;
+
+const DeployButton = styled.button`
+  background: ${({ theme }) => theme.color.background.deployButton};
+  color: ${({ theme }) => theme.color.text.listItemQuickButtonPrimary};
+  position: absolute;
+  font-weight: bold;
+  right: 28px;
+  border: none;
+  cursor: pointer;
+  border-radius: 10px;
+  height: 30px;
+  width: 80px;
+`;
+
+interface DeployChain {
+  chainId: number;
+  status?: string;
+  title: string;
+  iconUrl?: string;
+  type?: string;
+  state: AccountStates;
+}
+
+const Deployment = () => {
+  const [deployedChains, setDeployedChains] = useState<DeployChain[]>([]);
+  const [undeployedChains, setUndeployedChains] = useState<DeployChain[]>([]);
+  const [confirmModal, setConfirmModal] = useState<boolean>(false);
+  const [selectedChain, setSelectedChain] = useState<number | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const { getSdkForChainId, accountAddress } = useEtherspot();
+  const theme: Theme = useTheme();
+
+  const fetchDeploymentData = async (chainId: number) => {
+    const sdk = getSdkForChainId(chainId);
+    if (accountAddress && sdk) {
+      try {
+        await sdk.computeContractAccount();
+        return await sdk.getAccount();
+      } catch (err) {
+        //
+      }
+    }
```

Comment:
Just a preference so that there are fewer indents
```suggestion
    if (!accountAddress || !sdk) return;
    
      try {
        await sdk.computeContractAccount();
        return await sdk.getAccount();
      } catch (err) {
        //
      }
```